### DISCUSSION TITLE
Update usage and imports across project when refactoring object names

### DISCUSSION THREAD

**Discussion by foster999** [microsoft/pylance-release#2617] [Enhancement]
Currently, the python plugin refactor supports the renaming of objects and their references within a single file. I'd like to be able to update the references/usage of an object or attribute across the whole project.

In my opinion, the most benefit would come from renaming usage and import statements across the project. However, [the equivalent refactor in PyCharm](https://www.jetbrains.com/help/pycharm/rename-refactorings.html#examples) includes a number of other "nice to have" options, including renaming tests and comments that reference the object.

**Comment by judej** [comment]
@foster999, do you mean across workspaces? the rename should already be working in the same project.

**Reply by foster999** [reply]
This was within workspaces, which I've recently noticed is now supported - and works like a charm 🙂

Other functionality above is not included, such as renaming associated test functions (following the test_f standard pattern), or references to the object within docstrings. I appreciate these are less simple and I'm not sure if there's much of a calling for them

**Reply by judej** [reply]
Thanks for the response. Currently there are no plans to support renaming test functions.

**Comment by heejaechang** [comment]
we are still working on a new ability which move top level symbols between files in same workspace and update all references to it automatically. aka, Move symbols.

**Reply by foster999** [reply]
Thanks, this is the remaining feature that would help most with my refactoring

**Comment by judej** [comment]
Closing discussion since this is already working.
